gremlin-go/driver/httpProtocol.go

```diff
@@ -0,0 +1,148 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package gremlingo
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+)
+
+type httpProtocol struct {
+	*protocolBase
+
+	serializer serializer
+	logHandler *logHandler
+}
+
+// waits for response, deserializes and processes results
+// function name is readLoop but is not actually a loop - just keeping the name due to the protocol interface for now
+func (protocol *httpProtocol) readLoop(resultSets *synchronizedMap, errorCallback func()) {
+	msg, err := protocol.transporter.Read()
+
+	// Deserialize response message
+	fmt.Println("Deserializing response")
+	resp, err := protocol.serializer.deserializeMessage(msg)
+	if err != nil {
+		protocol.logHandler.logf(Error, logErrorGeneric, "httpReadLoop()", err.Error())
+		readErrorHandler(resultSets, errorCallback, err, protocol.logHandler)
+		return
+	}
+
+	fmt.Println("Deserialized response")
+	// TODO we should not need to use response/request ids to correlate responses with requests anymore after moving from ws to http
+	// but for simplicity of http POC we are just setting the responseId to the requestId here
+	resp.responseID = protocol.request.requestID
+	err = protocol.responseHandler(resultSets, resp)
+	if err != nil {
+		readErrorHandler(resultSets, errorCallback, err, protocol.logHandler)
+		return
+	}
+}
+
+func newHttpProtocol(handler *logHandler, url string, connSettings *connectionSettings) (protocol, error) {
+	transport, err := getTransportLayer(Http, url, connSettings, handler)
+	if err != nil {
+		return nil, err
+	}
+
+	gremlinProtocol := &httpProtocol{
+		protocolBase: &protocolBase{transporter: transport},
+		serializer:   newGraphBinarySerializer(handler),
+		logHandler:   handler,
+	}
+	return gremlinProtocol, nil
+}
+
+// loads results into the response set from the response
+func (protocol *httpProtocol) responseHandler(resultSets *synchronizedMap, response response) error {
+	fmt.Println("Handling response")
+
+	// TOPO http specific response handling - below is just copy-pasted from web socket implementation for now
```

Comment:
Is this a typo or an acronym I don't know, want to make sure any TODO's are searchable in the future.
```suggestion
	// TODO http specific response handling - below is just copy-pasted from web socket implementation for now
```

Reply:
Lol it is a typo! Will fix it